### ISSUE TITLE
Tuple types

### ISSUE THREAD

**Issue by thosakwe** [bonobo-lang/bonobo#8]
The `()` can have a different meaning when multiple expressions are contained.

```bonobo
(1.0) // 1.0
(1.0, 2.0) // (Num, Num)
(1, '2', 3.0) // (Num, String, Num)
```

This will play nicely with typedefs.

**Comment by thosakwe** [comment]
How to access the members of a tuple? Probably similar to Python.

```bonobo
f main {
  v tup := (1.0, '2.0')
  print tup[0] // 1.0
  print tup[1] // '2.0'
  ret tup[0:1] // Produces (Num, String)
}
```

**Comment by thosakwe** [comment]
What if tuples didn't even need parentheses? That would just mean that call expressions would take a tuple instead of a list of arguments.

Which would make the following *legal*:

```bonobo
f add (a: Num, b: Num) => a + b

f seven => add 3, 4
```

**Comment by thosakwe** [comment]
The last things left:
1. Parse tuple type declarations from text
2. Compile them into custom struct

**Comment by thosakwe** [comment]
Closed, parsing is targeted in #13.
